packages/relay/src/lib/services/metricService/metricService.ts

```diff
@@ -113,7 +113,7 @@ export default class MetricService {
     this.consensusNodeClientHistogramGasFee = this.initGasMetric(register);
 
     this.eventEmitter.on(constants.EVENTS.EXECUTE_TRANSACTION, (args: IExecuteTransactionEventPayload) => {
-      this.captureTransactionMetrics(args);
+      this.captureTransactionMetrics(args).then();
```

Comment:
qq: why do we need a .then() here?

Reply:
If you're calling an async function without awaiting it, Sonar suggests adding a `.then()` or `.catch()` to make it explicit that the function is intentionally not awaited. This signals to other developers (or to static analysis tools like Sonar) that you are aware of the behaviour and that this is intentional. Otherwise, they would flag a warning for “missing await on async function”.

Reply:
Gotcha thanks for clarifying!